ecl/hql/hqlexpr.cpp

```diff
@@ -7944,10 +7944,10 @@ void CHqlScope::throwRecursiveError(IIdAtom * searchName)
     StringBuffer filename;
     if (fullName)
         filename.append(fullName).append('.');
-    filename.append(*searchName);
```

Comment:
This change looks suspicious.  It should be 
   filename.append(str(searchName));
or even
   filename.append(searchName)

Previously the operator * was defined so that *myatom was the same as myatom->str();

*str(myAtom) will return just the 1st character